security/advancedtls/crl_test.go

```diff
@@ -731,3 +731,35 @@ func TestIssuerNonPrintableString(t *testing.T) {
 		t.Fatalf("fetchCRL failed: %s", err)
 	}
 }
+
+func TestCRLCacheExpirationReloading(t *testing.T) {
+	cache, err := lru.New(5)
+	if err != nil {
+		t.Fatalf("Creating cache failed")
+	}
+
+	var certs = makeChain(t, testdata.Path("crl/revokedInt.pem"))
+	// Certs[1] has the same issuer as the revoked cert
+	rawIssuer := certs[1].RawIssuer
+
+	// `3.crl`` revokes `revokedInt.pem`
+	crl := loadCRL(t, testdata.Path("crl/3.crl"))
+	// Modify the crl so that the cert is NOT revoked and add it to the cache
+	crl.CertList.TBSCertList.RevokedCertificates = nil
+	crl.CertList.TBSCertList.NextUpdate = time.Now().Add(time.Hour)
+	cache.Add(hex.EncodeToString(rawIssuer), crl)
+	var cfg = RevocationConfig{RootDir: testdata.Path("crl"), Cache: cache}
+	revocationStatus := checkChain(certs, cfg)
+	if revocationStatus != RevocationUnrevoked {
+		t.Fatalf("Certificate check should be RevocationUnrevoked, was %v", revocationStatus)
+	}
+
+	// Modify the entry in the cache so that the cache will be refreshed
+	crl.CertList.TBSCertList.NextUpdate = time.Now()
+	cache.Add(hex.EncodeToString(rawIssuer), crl)
```

Comment:
I am a bit confused here....if we are manipulating the cache directly, what behaviors are we testing against?(since theoretically speaking you can change it to anything you want if you take control of the cache)

Reply:
This is a way for us to fake reaching the `NextUpdate` time for the cached CRL. Otherwise, we would have to do a more complicated mocking of `time` for fast-forward to the time at which the CRL's `NextUpdate` field has passed.